packages/data/src/components/use-select/index.js

```diff
@@ -181,8 +187,18 @@ function Store( registry, suspense ) {
 
 		lastIsAsync = isAsync;
 
+		// Check if the store changed between the `updateValue` call above and
+		// the actual `subscribe` call.
+		const cleanup = subscriber.subscribe( () => {
+			lastMapResultValid = false;
+		} );
+		const subscribe = ( _listener ) => {
+			cleanup();
+			return subscriber.subscribe( _listener );
+		};
```

Comment:
If the returned `subscribe` function is never called, the subscription you’re newly creating will leak, if will be there forever and never cleaned up.

Reply:
Will fix in the next commit

Reply:
@jsnajdr Ok, I changed it to avoid subscribing and simply compare the store states with those on mount. If all the states are equal, then `lastMapResultValid` can remain true.